capstone-backend/app.js

```diff
@@ -142,28 +184,34 @@ async function getMatches(params, currentUser) {
         const matchQuery = new Parse.Query(Match);
         matchQuery.equalTo("user_1", currentUser.id);
         matchQuery.equalTo("user_2", entry.id);
-        let matchResults = await matchQuery.find();
+        let matchResults = await matchQuery.first();
 
+        const matchQuery2 = new Parse.Query(Match);
+        matchQuery2.equalTo("user_2", currentUser.id);
+        matchQuery2.equalTo("user_1", entry.id);
+        let matchResults2 = await matchQuery2.first();
 
-        if (matchResults.length > 0) {
-            for (let i = 0; i < matchResults.length; i++) {
-                //update match score if needed
-                matchResults[i].set("score", matchScore);
-                if (params.matchId == matchResults[i].get("user_2")) {
-                    matchResults[i].set("liked", params.liked);
-                }
-                await matchResults[i].save();
-            }
+        if (matchResults) {
+            matchResults.set("score", matchScore);
+            matchResults2.set("score", matchScore);
         }
         else {
             const match = new Match();
+            const match2 = new Match();
             if (matchScore) {
                 match.set("score", matchScore);
                 match.set("liked", false);
-                match.set("seen", false);
                 match.set("user_1", currentUser.id);
                 match.set("user_2", entry.id);
+                match.set("display_private", false);
                 await match.save();
```

Comment:
you can refractor that into a helper function ,since you will be doing the same thing right after this with the other user.